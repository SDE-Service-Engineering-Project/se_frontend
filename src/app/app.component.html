<app-header *ngIf="isLoggedIn()"></app-header>
<app-toast aria-live="polite" aria-atomic="true"></app-toast>
<router-outlet></router-outlet>
