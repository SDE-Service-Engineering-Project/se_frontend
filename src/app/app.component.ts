import {Component} from '@angular/core';
import {ACCESS_TOKEN, StorageService} from "./services/storage.service";

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.sass']
})
export class AppComponent {
  title = 'CarRental';


  constructor(private storageService: StorageService) {
  }

  isLoggedIn() {
    return !!this.storageService.getItem(ACCESS_TOKEN)
  }
}
