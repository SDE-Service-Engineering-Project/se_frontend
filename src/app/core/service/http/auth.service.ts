import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {AuthResponse} from "../../model/AuthResponse";

const AUTH_API = 'http://localhost:8080/auth/';

@Injectable({
  providedIn: 'root'
})
export class AuthService {

  constructor(private http: HttpClient) {
  }

  public login(userName: string, password: string): Observable<AuthResponse> {
    return this.http.post<AuthResponse>(
      AUTH_API + 'login',
      {userName, password}
    );
  }
}
