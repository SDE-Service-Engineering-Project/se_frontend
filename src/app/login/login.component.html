<div class="container-fluid center-screen">
  <div id="main-wrapper" class="container">
    <div class="row justify-content-center">
      <div class="col-xl-10">
        <div class="card border-0">
          <div class="card-body p-0">
            <div class="row no-gutters">
              <div class="col-lg-6">
                <div class="p-5">
                  <div class="mb-5">
                    <h3 class="h4 font-weight-bold text-theme" style="color: #5369f8">Login</h3>
                  </div>
                  <h6 class="h5 mb-0">Welcome back!</h6>
                  <p class="text-muted mt-2 mb-5">Enter your user name and password to access the best cars in the
                    world.</p>

                  <form [formGroup]="loginForm" id="loginForm" (ngSubmit)="submitLogin()" >
                    <div class="form-group" style="text-align: left">
                      <label for="inputUserName">Username</label>
                      <input type="text" class="form-control" id="inputUserName" formControlName="username"
                             [ngClass]="{ 'is-invalid': loginForm.get('username')?.touched && loginForm.get('username')!!.errors, 'is-valid': loginForm.get('username')!!.valid }"
                      >
                      <div *ngIf="loginForm.get('username')!!.touched && loginForm.get('username')!!.errors"
                           class="invalid-feedback">
                        <div *ngIf="loginForm.get('username')!!.errors!!.hasOwnProperty('required')" id="err-username">Username is required!
                        </div>
                      </div>
                    </div>

                    <div class="form-group mb-5 mt-2" style="text-align: left">
                      <label for="inputPassword">Password</label>
                      <input type="password" class="form-control" id="inputPassword" formControlName="password"
                             [ngClass]="{ 'is-invalid': loginForm.get('password')!!.touched && loginForm.get('password')!!.errors, 'is-valid': loginForm.get('password')!!.valid }"
                      >
                      <div *ngIf="loginForm.get('password')!!.touched && loginForm.get('password')!!.errors"
                           class="invalid-feedback">
                        <div *ngIf="loginForm.get('password')!!.errors!!.hasOwnProperty('required')" id="err-password">Password is
                          required!
                        </div>
                      </div>
                    </div>
                    <div class="form-group mb-5">
                      <button type="submit" class="btn btn-theme form-control" id="login-btn" [disabled]="loginForm!!.invalid">Login</button>
                    </div>
                  </form>
                </div>
              </div>
              <div class="col-lg-6 d-none d-lg-inline-block">
                <div class="logo rounded-right">
                  <div class="overlay rounded-right"></div>
                  <div class="logo-inner">
                    <h4 class="text-white mb-4">Welcome to Car Rental</h4>
                    <p class="lead text-white">"Where cars are still cars."</p>
                    <p>- Godfather (CEO)</p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <!--TODO: add register component-->
        <p class="text-muted text-center mt-3 mb-0">Don't have an account? Register <a
          href="https://www.youtube.com/watch?v=dQw4w9WgXcQ" target="_blank" rel="noopener" class="text-primary ml-1">here</a>.
        </p>
      </div>
    </div>
  </div>
</div>
