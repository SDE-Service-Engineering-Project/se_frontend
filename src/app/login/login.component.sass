.logo
  padding: 0
  background-image: url(https://media.tenor.com/d0BeKMg4ohYAAAAC/car.gif)
  background-repeat: no-repeat
  background-size: cover
  height: 100%
  position: relative
  .overlay
    -webkit-box-flex: 1
    -ms-flex: 1
    flex: 1
    position: absolute
    top: 0
    bottom: 0
    left: 0
    right: 0
    background-color: rgba(0, 0, 0, 0.4)

  .logo-inner
    text-align: center
    color: #fff
    position: absolute
    margin: 0 auto
    padding: 0 1.75rem
    bottom: 3rem
    left: 0
    right: 0

.btn-theme
  background-color: #5369f8
  border-color: #5369f8
  color: #fff

.center-screen
  display: flex
  justify-content: center
  align-items: center
  text-align: center
  min-height: 100vh
