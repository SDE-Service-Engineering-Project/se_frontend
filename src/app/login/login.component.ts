import {Component, OnInit} from '@angular/core';
import {FormControl, FormGroup, Validators} from "@angular/forms";

@Component({
  selector: 'app-login',
  templateUrl: './login.component.html',
  styleUrls: ['./login.component.sass']
})
export class LoginComponent implements OnInit {
  loginForm = new FormGroup({
    username: new FormControl("", [Validators.required]),
    password: new FormControl("", [Validators.required])
  });


  ngOnInit(): void {

  }


  submitLogin() {
    if (this.loginForm.invalid) return;

    console.log("make login call user:" + this.loginForm.get('username')!!.value + " password: " + this.loginForm.get('password')!!.value)
  }
}
