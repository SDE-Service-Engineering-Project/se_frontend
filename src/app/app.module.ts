import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppRouting } from './app.routing';
import { AppComponent } from './app.component';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { SharedModule } from './shared/shared.module';
import { httpInterceptor } from './services/interceptor/http.interceptor';
import { HttpClientModule } from '@angular/common/http';

@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    AppRouting,
    NgbModule,
    SharedModule,
    HttpClientModule,
  ],
  providers: [httpInterceptor],
  bootstrap: [AppComponent],
})
export class AppModule {}
