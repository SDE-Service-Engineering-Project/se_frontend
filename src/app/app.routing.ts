import {NgModule} from '@angular/core';
import {RouterModule, Routes} from '@angular/router';
import {AuthGuard} from "./services/guards/auth.guard";
import {LoginGuard} from "./services/guards/login.guard";

const routes: Routes = [
  {
    path: '',
    canActivate: [LoginGuard],
    loadChildren: () =>
      import('./modules/login/login.module').then((m) => m.LoginModule),
  },
  {
    path: 'register',
    canActivate: [LoginGuard],
    loadChildren: () => import('./modules/register/register.module').then((m) => m.RegisterModule),
  },
  {
    path: 'home',
    loadChildren: () =>
      import('./modules/landing-page/landing-page.module').then(
        (m) => m.LandingPageModule
      ),
    canActivate: [AuthGuard]
  },
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule],
})
export class AppRouting {
}
