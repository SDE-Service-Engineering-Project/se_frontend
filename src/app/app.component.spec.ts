import {ComponentFixture, TestBed} from '@angular/core/testing';
import {AppComponent} from './app.component';
import {createComponentFactory, Spectator} from '@ngneat/spectator/jest';
import {RouterTestingModule} from '@angular/router/testing';
import {LoginComponent} from "./login/login.component";
import {HttpClientTestingModule} from "@angular/common/http/testing";

describe('AppComponent', () => {
  let spectator: Spectator<AppComponent>;
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  const createComponent = createComponentFactory({
    detectChanges: false,
    component: AppComponent,
    imports: [RouterTestingModule],
  });

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [HttpClientTestingModule],
      declarations: [AppComponent, LoginComponent]
    }).compileComponents()
    spectator = createComponent();
    fixture = spectator.fixture;
    component = spectator.component;
  });

  it('should render the app', () => {
    expect(fixture.debugElement.nativeElement).toMatchSnapshot();
  });

  it(`should display login page`, () => {
    expect(component.title).toEqual('CarRental');
  });
});
