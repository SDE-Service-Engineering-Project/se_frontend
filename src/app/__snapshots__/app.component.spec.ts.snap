// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render the app 1`] = `
<div
  id="root0"
>
  
  <app-toast
    aria-atomic="true"
    aria-live="polite"
  />
  <router-outlet />
  
</div>
`;
